torchtitan/config_manager.py

```diff
@@ -532,10 +532,22 @@ def _args_to_two_level_dict(self, args: argparse.Namespace) -> defaultdict:
             args_dict[first_level_key][second_level_key] = v
         return args_dict
 
-    def _validate_config(self) -> bool:
+    def _validate_config(self) -> None:
         # TODO: Add more mandatory validations
-        assert self.model.name and self.model.flavor and self.model.tokenizer_path
-        return True
+        assert self.model.name
+        assert self.model.flavor
+        assert self.model.tokenizer_path
+
+        ac_config = self.activation_checkpoint
```

Comment:
This part is worth discussion @tianyu-l .

I do think that we want checks as early as possible (like a post-init validation of the `JobConfig`) so that we avoid having to build device mesh/dataloader before erroring. (Also refer to https://github.com/pytorch/torchtitan/pull/449.)

On the flip side, these checks are now not local to where the parallelism/technique gets applied, so the invariants are not as obvious.

I am okay to leave these checks in `parallelize_llama.py` if you think that is better.